neo/SmartContract/InteropService.cs

```diff
@@ -546,16 +546,30 @@ private static bool PutEx(ApplicationEngine engine, StorageContext context, byte
             if (value.Length > MaxStorageValueSize) return false;
             if (context.IsReadOnly) return false;
             if (!CheckStorageContext(engine, context)) return false;
+
             StorageKey skey = new StorageKey
             {
                 ScriptHash = context.ScriptHash,
                 Key = key
             };
-            StorageItem item = engine.Snapshot.Storages.GetAndChange(skey, () => new StorageItem());
-            if (item.IsConstant) return false;
-            item.Value = value;
-            item.IsConstant = flags.HasFlag(StorageFlags.Constant);
-            return true;
+
+            StorageItem item = engine.Snapshot.Storages.TryGet(skey);
+            if (item?.IsConstant == true) return false;
+
+            if (value.Length == 0)
+            {
+                // If is empty, we try to remove it
+
+                engine.Snapshot.Storages.Delete(skey);
+                return true;
```

Comment:
If I put an empty bytes with `StorageFlags.Constant` flag, what will happen?

Reply:
The expected behaviour is: write nothing, read empty bytes

Reply:
SInce `Get` already returns empty array, we shouldn't keep empty values.. even if constant.

Reply:
What's an example use case for `const`?

Reply:
Only explicit `Delete` will destroy them, right @erikzhang ? I just noticed `Delete` cannot reap const values too. So these are contract-lifetime values, correct?